tests/show_techsupport/test_auto_techsupport.py

```diff
@@ -836,40 +840,34 @@ def validate_techsupport_generation(duthost, is_techsupport_expected, expected_c
     :param since_value_in_seconds: int, value in seconds which used in validation for since parameter
     :return: AssertionError in case of failure
     """
-    expected_oldest_timestamp_datetime = None
+    expected_oldest_log_line_timestamps_list = None
     if since_value_in_seconds:
-        expected_oldest_timestamp_datetime = get_expected_oldest_timestamp_datetime(duthost, since_value_in_seconds)
+        expected_oldest_log_line_timestamps_list = get_expected_oldest_timestamp_datetime(duthost,
+                                                                                          since_value_in_seconds)
 
     try:
         available_tech_support_files = duthost.shell('ls /var/dump/*.tar.gz')['stdout_lines']
     except RunAnsibleModuleFail:
         available_tech_support_files = []
 
     if is_techsupport_expected:
-        expected_techsupport_files_num = 1
+        expected_techsupport_files = True
         assert is_techsupport_generation_in_expected_state(duthost, expected_in_progress=True), \
             'Expected techsupport generation not started or expected number of processes does not match actual number'
         wait_until(300, 10, 0, is_techsupport_generation_in_expected_state, duthost, False)
     else:
         assert is_techsupport_generation_in_expected_state(duthost, expected_in_progress=False), \
             'Unexpected techsupport generation in progress'
-        expected_techsupport_files_num = 0
+        expected_techsupport_files = False
 
-    try:
-        new_available_tech_support_files = duthost.shell('ls /var/dump/*.tar.gz')['stdout_lines']
-    except RunAnsibleModuleFail:
-        new_available_tech_support_files = []
-    new_techsupport_files_list = list(set(new_available_tech_support_files) - set(available_tech_support_files))
-    new_techsupport_files_num = len(new_techsupport_files_list)
-    logger.info('New files created: {} expected: {}'.format(new_techsupport_files_num,
-                                                            expected_techsupport_files_num))
-    assert new_techsupport_files_num == expected_techsupport_files_num, \
-        'Number of new techsupport files {} does not match number of expected {} files'.format(
-            new_techsupport_files_num,
-            expected_techsupport_files_num)
+    if expected_techsupport_files:
+        # techsupport file creation may took some time after generate dump process already finished
+        assert wait_until(300, 10, 0, is_new_techsupport_file_generated, duthost, available_tech_support_files), \
```

Comment:
will it wait the max time or it will poll? want to reduce the test time if possible

Reply:
it will poll every 10 sec, up to 300 sec